runtime/src/accounts.rs

```diff
@@ -153,12 +152,7 @@ impl Accounts {
             // If a fee can pay for execution then the program will be scheduled
             let mut accounts: TransactionAccounts = Vec::with_capacity(message.account_keys.len());
             let mut tx_rent: TransactionRent = 0;
-            for (i, key) in message
-                .account_keys
-                .iter()
-                .enumerate()
-                .filter(|(i, key)| Self::is_non_loader_key(message, key, *i))
-            {
+            for (i, key) in message.account_keys.iter().enumerate() {
```

Comment:
It will change who the payer account is which breaks compatibility with the older way of processing transactions.

Reply:
if `account_keys[0]` is filtered by `Self::is_non_loader_key(message, key, *i)`, then `accounts[0]` will be a different account than it was previously.